RTCCertificate fixes (#27107)
mdn/content: files/en-us/web/api/rtccertificate/getfingerprints/index.md

---
title: "RTCCertificate: getFingerprints() method"
short-title: getFingerprints()
slug: Web/API/RTCCertificate/getFingerprints
page-type: web-api-instance-method
browser-compat: api.RTCCertificate.getFingerprints
---

{{APIRef("WebRTC")}}

> The **`getFingerprints()`** method of the **{{domxref("RTCCertificate")}}** interface is used to get the array of certificate fingerprints.

Reviewer: ```suggestion
The **`getFingerprints()`** method of the **{{domxref("RTCCertificate")}}** interface is used to get an array of certificate fingerprints.
```

> An application can compare these certificate fingerprints from the client to certificates from the server.

Reviewer: ```suggestion
An application can use this method to compare the client certificate fingerprints with the certificate fingerprints from the server.
```

The server and client may support different sets of algorithms: all fingerprint values for the set of algorithms supported by both client and server should match.

## Syntax

```js-nolint
getFingerprints()
```

### Parameters

None

### Return value

An [Array](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) of fingerprint values.
Each fingerprint is represented by an object with the following properties:

- `algorithm`
  - : A string indicating the hash function algorithm used to create the fingerprint in `value`.

>     Allowed values include: `"sha-1"`, `"sha-224"`, `"sha-256"`,`"sha-384"`, `"sha-512"`, `"md5"`, `"md2"`.<!-- from [RFC4572] Section 5.>

Reviewer 2: ```suggestion
    Allowed values include: `"sha-1"`, `"sha-224"`, `"sha-256"`,`"sha-384"`, `"sha-512"`, `"md5"`, `"md2"`.<!-- from [RFC4572] Section 5. -->
```

Author: Thank you!

---
title: "RTCCertificate: getFingerprints() method"
short-title: getFingerprints()
slug: Web/API/RTCCertificate/getFingerprints
page-type: web-api-instance-method
browser-compat: api.RTCCertificate.getFingerprints
---

{{APIRef("WebRTC")}}

The **`getFingerprints()`** method of the **{{domxref("RTCCertificate")}}** interface is used to get the array of certificate fingerprints.

An application can compare these certificate fingerprints from the client to certificates from the server.

> The server and client may support different sets of algorithms: all fingerprint values for the set of algorithms supported by both client and server should match.

Author: FYI it was hard to determine exactly how these fingerprints are used, and the specs were not much use. After a little chatGPT, a lot of googling, and what I think is common sense, I am fairly sure that this is correct.

Essentially a fingerprint is a hash of a certificate with a particular algorithm. A webrtc server sends fingerprints for the certificate using the algorithms it supports during TLS setup. The client can calculate the certificate fingerprints using the algorithms it supports. It is not clear to me how the client gets the server certificates, but it is clear that if it can get them it can compare the set of common-algorithm certificate hashes, and if they all match it can trust the server has matching certificates. 

It is a bit similar to WebTransport, but there the API is supplied a set of hash values and the browser compares these to the hash it calculates from the received certificate. Here the client might generate the certificate or be passed a certificate so it is easy to see how it might calculate the hash. However it is not clear to me how this all fits together, because if the peer connection is creating a certificate, then how is the server getting the certificate???

Posting because I hope you know more!
